src/AlevinUtils.cpp

```diff
@@ -570,6 +627,51 @@ namespace alevin {
         std::not2(std::equal_to<std::string::value_type>()));
     }
 
+    void addPadding(std::string& seq, uint32_t max, std::string padBases, unsigned int padLen) {
+      int diff = max - seq.length() + 1; // add one base if the length is same to avoid erroneous collisions
+      int rep = diff / padLen;
+      int extra = diff % padLen;
+      for(int i = 0; i < rep; i++){
+          seq += padBases;
+      }
+      seq += padBases.substr(0,extra);
+    }
+
+    void modifyRegex(size_t readNumber, std::string type, std::string* reg, std::vector<int> *bu, unsigned int& nPat, std::size_t first, std::size_t second)
```

Comment:
Is `type` always a single character?  If so, we should probably avoid using a string for it.  A character would be better.  Better yet, if it always takes on one of a small set of values, we should make it an `enum class`.